src/main/java/uk/ac/ic/wlgitbridge/bridge/util/Pair.java

```diff
@@ -0,0 +1,39 @@
+package uk.ac.ic.wlgitbridge.bridge.util;
+
+import java.util.function.Function;
+
+public class Pair<L, R> {
```

Comment:
Just a note, I couldn't find a Pair/Tuple type in our current dependencies (or, at least IntelliJ couldn't), and the internet suggested this did not exist in the standard library, so I wrote a custom class.